internal/cmd/buildtool/generic.go

```diff
@@ -55,3 +64,17 @@ func genericBuildPackage(deps buildtoolmodel.Dependencies, product *product) {
 
 	runtimex.Try0(shellx.RunEx(defaultShellxConfig(), argv, &shellx.Envp{}))
 }
+
+// genericBuildLibrary is the generic function for building a library.
+func genericBuildLibrary(deps buildtoolmodel.Dependencies, product *product) {
+	// deps.GolangCheck()
+
+	log.Infof("building %s for %s/%s", product.Pkg, runtime.GOOS, runtime.GOARCH)
+
+	argv := runtimex.Try1(shellx.NewArgv("go", "build"))
+	argv.Append("-buildmode", "c-shared")
+	argv.Append("-o", fmt.Sprintf("%s.so", product.Pkg)) // Note: we are only building for linux for now
```

Comment:
We will need later a more general function that maps a `runtime.GOOS` to the expected library extension. I suggest we already create such a function and use it to set the extension here.